diff --git a/node_modules/@sentry/react-native/RNSentry.podspec b/node_modules/@sentry/react-native/RNSentry.podspec
index 2d9950f..c0d8899 100644
--- a/node_modules/@sentry/react-native/RNSentry.podspec
+++ b/node_modules/@sentry/react-native/RNSentry.podspec
@@ -8,7 +8,7 @@ Pod::Spec.new do |s|
   s.summary        = 'Official Sentry SDK for react-native'
   s.author         = 'Sentry'
   s.homepage       = "https://github.com/getsentry/sentry-react-native"
-  s.source         = { :git => 'https://github.com/getsentry/sentry-react-native.git', :tag => "#{s.version}"}
+  s.source         = { :git => 'https://github.com/getsentry/sentry-cocoa.git', :tag => "#{s.version}"}
 
   s.ios.deployment_target = "8.0"
   s.osx.deployment_target = "10.10"
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.preserve_paths = '*.js'
 
   s.dependency 'React-Core'
-  s.dependency 'Sentry', '7.11.0'
+  s.dependency 'Sentry', '7.16.0'
 
   s.source_files = 'ios/RNSentry.{h,m}'
   s.public_header_files = 'ios/RNSentry.h'
