diff --git a/node_modules/metro-runtime/src/polyfills/require.js b/node_modules/metro-runtime/src/polyfills/require.js
index e7e1e25..2b4068c 100644
--- a/node_modules/metro-runtime/src/polyfills/require.js
+++ b/node_modules/metro-runtime/src/polyfills/require.js
@@ -110,14 +110,20 @@ function metroRequire(moduleId) {
     if (initializingIndex !== -1) {
       const cycle = initializingModuleIds
         .slice(initializingIndex)
-        .map((id) => (modules[id] ? modules[id].verboseName : "[unknown]")); // We want to show A -> B -> A:
+        .map((id) => (modules[id] ? modules[id].verboseName : "[unknown]"));
 
+      // We want to show A -> B -> A: do this for our own code
+      const isExternalOnly = cycle.every(function (cycleWarning) {
+        return cycleWarning.includes("node_modules");
+      });
+      if (!isExternalOnly) {
       cycle.push(cycle[0]);
       console.warn(
         `Require cycle: ${cycle.join(" -> ")}\n\n` +
           "Require cycles are allowed, but can result in uninitialized values. " +
           "Consider refactoring to remove the need for a cycle."
       );
+      }
     }
   }
 
